### DISCUSSION TITLE
refresh authToken before user logged get out due inactivity

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#4044] [Q&A]
Users of my website complain that they are logged out every 6 hours. That makes sense, because after 6 hours the JWT token expires. I'm looking for a way when they try to make a request that when the token expires it doesn't log them out but gives them a new token and proceed the request. 

What can I do when req.pbClient.authStore.isValid is false? Does authRefresh works?

````
const pocketInstance = require('pocketbase/cjs');

verifyToken = async (req, res, next) => {
    req.pbClient = new pocketInstance(process.env.POCKETBASE_SERVER);

    req.pbClient.authStore.loadFromCookie(req.headers.cookie || '');

    req.pbClient.authStore.onChange(() => {
        res.setHeader('Set-Cookie', req.pbClient.authStore.exportToCookie({ httpOnly: false }));
    });

    try {
        const response = req.pbClient.authStore.isValid;

        if (!response) {
            return res.status(401).send({
                message: 'Your session has expired due to inactivity. You will be redirected to the login page.',
            });
        }

        req.user = req.pbClient.authStore.baseModel;
    } catch (_) {
        req.pbClient.authStore.clear();
        next(_);
    }

    next();
};

module.exports = {
    verifyToken,
};
````

**Comment by ganigeorgiev** [comment]
>  when the token expires it doesn't log them out but gives them a new token and proceed the request.

I'm not sure that I understand what do you mean by the above. You either have expiration or not. If you want to have longer tokens expiration just change the default tokens duration time from the _Admin UI > Settings > Token options_.

> What can I do when req.pbClient.authStore.isValid is false? Does authRefresh works?

`req.pbClient.authStore.isValid` doesn't perform any requests. It just loosely checks whether the current instance store is populated and has non-expired token.
Call `authRefresh()` if you want to validate and renew the user token with each request, eg.:
```js
...

try {
    // get an up-to-date auth store state by verifying and refreshing the loaded auth model (if any)
    req.pbClient.authStore.isValid && await req.pbClient.collection('users').authRefresh();
} catch (_) {
    // clear the auth store on failed refresh
    req.pbClient.authStore.clear();
}

...
```
